uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/ConfigurationsEditorBeanTest.java

```diff
@@ -282,6 +511,170 @@ public void testSettingBeanProperties() {
+		methodDialog.cancel();
+		editor.close(true);
+
+		assertEquals("Only public static methods should be listed as possible factory methods",
+				asList("publicStaticVoid()", "publicStaticFactory()"), actualMethods);
+	}
+
+	/**
+	 * <p>
+	 * Tests the list of factory methods in a bean referencing a factory bean.
+	 * </p>
+	 * <b>Steps:</b>
+	 * <ol>
+	 * <li>Open Properties view
+	 * <li>Open a camel context in Camel editor
+	 * <li>Switch to Configurations
+	 * <li>Select bean 'helloBean'
+	 * <li>Select factory bean 'factoryBean'
+	 * <li>Browse factory methods
+	 * <li>Check if the list contains only public non-static methods from 'FactoryBean'
+	 * </ol>
+	 */
+	@Test
+	public void testListingReferenceFactoryMethods() {
+		ConfigurationsPropertiesView properties = new ConfigurationsPropertiesView();
+		properties.open();
+
+		new CamelProject(projectName).openCamelContext(projectType.getCamelContext());
+		editor = new ConfigurationsEditor();
+		editor.selectBean("helloBean");
+		properties.activate();
+		properties.toggleBeanReference();
+		properties.selectBeanReference("factoryBean");
+
+		List<String> actualMethods = new ArrayList<>();
+		MethodSelectionDialog methodDialog = properties.browseFactoryMethod();
+		methodDialog.getItems().stream().forEach(item -> actualMethods.add(item.getText()));
+		methodDialog.cancel();
+		editor.close(true);
+
+		assertEquals("Only public non-static methods should be listed as possible reference factory methods",
+				asList("publicCreateHelloBean()"), actualMethods);
+	}
+
+	/**
+	 * <p>
+	 * Tests the list of init methods in a bean.
+	 * </p>
+	 * <b>Steps:</b>
+	 * <ol>
+	 * <li>Open Properties view
+	 * <li>Open a camel context in Camel editor
+	 * <li>Switch to Configurations
+	 * <li>Select bean 'helloBean'
+	 * <li>Select Java class 'HelloBean'
+	 * <li>Browse init methods
+	 * <li>Check if the list contains only public void parameter-free methods from 'HelloBean'
+	 * </ol>
+	 */
+	@Test
+	public void testListingInitMethods() {
+		ConfigurationsPropertiesView properties = new ConfigurationsPropertiesView();
+		properties.open();
+
+		new CamelProject(projectName).openCamelContext(projectType.getCamelContext());
+		editor = new ConfigurationsEditor();
+		editor.selectBean("helloBean");
+		properties.activate();
+		properties.toggleBeanClass();
+		FilteredSelectionDialog dialog = properties.browseClass();
+		dialog.setText("HelloBean");
+		dialog.waitForItems();
+		dialog.selectItem("HelloBean - hello");
+		dialog.ok();
+
+		Set<String> actualMethods = new HashSet<>();
+		MethodSelectionDialog methodDialog = properties.browseFactoryMethod();
+		methodDialog.getItems().stream().forEach(item -> actualMethods.add(item.getText()));
+		methodDialog.cancel();
+		editor.close(true);
+
+		assertEquals("Only public void parameter-free methods should be listed as possible init methods",
+				asSet("publicVoid()", "publicStaticVoid()"), actualMethods);
```

Comment:
you can use `java.util.Arrays.asList` here, can't you?

Reply:
I don't want to be so strict on the order of displayed method. That's why I use set instead of list.

Reply:
It makes sense. I overlooked that `actualMethods` variable is also `Set`